crawler/crawling/distributed_scheduler.py

```diff
@@ -563,11 +563,18 @@ def request_from_feed(self, item):
             req.meta[key] = item[key]
 
         # extra check to add items to request
+        if 'headers' in item and item['headers'] is not None:
+            for key, value in item['headers'].iteritems():
+                req.headers[key] = value
+
         if 'cookie' in item and item['cookie'] is not None:
-            if isinstance(item['cookie'], dict):
```

Comment:
This code was not getting executed because kafka scraper_schema.json is only forcing it to be string. So I added attribute `cookies` to scraper_schema. Not sure if it make sense.